Sources/ForageSDK/Foundation/Network/ForageAPI.swift

```diff
@@ -60,7 +60,8 @@ extension ForageAPI: ServiceProtocol {
                 "authorization": "Bearer \(model.authorization)",
                 "content-type": "application/json",
                 "accept": "application/json",
-                "API-VERSION": "2023-05-15"
+                "API-VERSION": "2023-05-15",
+                "x-datadog-trace-id": model.traceId ?? ""
```

Comment:
should we set this header on all HTTP requests with one line, rather than drilling it through every request?

```swift
// in Provider.swift or ServiceProtocol.swift
            request.setValue(traceId, forHTTPHeaderField: "x-datadog-trace-id")
```